### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by gregorbg**
Move over two subprojects from `thewca/tnoodle`.

This is mostly done so that Android can be removed from the main repository and installing the humongously large Android SDK is not a dev requirement for the core TNoodle project anymore.

The only things that changed are:
- Adapt README (duh.)
- Remove unused dependencies from the buildSrc dependency file.

Everything else is literal c/p from the TNoodle main repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+Every sub-project has its individual artifact configuration and `build.gradle` file. Furthermore, there is a central `buildSrc` folder,
+which is automatically sourced by Gradle. It contains common code and shared configuration setups.
+
+
+### tnoodle-android
+
+A sample Android app with a drawer that lets you choose between several puzzles and generates scrambles for them.
+
+Note that in order for this setup to compile, you will have to **install the Android SDK** on your local machine.
+Steps and instructions for this can be found in the main [TNoodle README](https://github.com/thewca/tnoodle/blob/master/README.md) and at [https://developer.android.com/studio](https://developer.android.com/studio)
```
Comment (jfly) [inline README.md]: Can we remove the android sdk instructions from the TNoodle readme and put them here instead?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    }
+
+    dependencies {
+        classpath(ANDROID_BUILD_TOOLS)
+    }
+}
+
+allprojects {
+    group = "org.worldcubeassociation.tnoodle-samples"
+    version = "0.14.0"
```
Comment (jfly) [inline build.gradle.kts]: Should we restart this version number so it's clear that it's unrelated to TNoodle core?

**Review by jfly — COMMENTED** [review]
Awesome, thanks! A few questions:

1. Is there a corresponding PR for removing this stuff from `thewca/tnoodle`?
2. Should we update the core TNoodle readme to reference this new repo in some way? I think that would be nice.
3. I think it would be nice to also set up travis to build this project. Doesn't have to happen as part of this PR.
4. I feel like the GWT compilation code doesn't belong in a repository named "samples". I'd prefer that samples is more like "here's a simple example of how to embed TNoodle in your application written in <insert language here>"
   - On that note: it feels strange to me that we're putting gradle stuff at the top level of this repository. I'd prefer a structure more like https://github.com/thewca/wca-oauth, where the top level of the repository is a README, and a folders correspoding to languages/frameworks.

**Comment by jfly** [comment]
Oh, one other thought: it might be worth going to the effort to pull the commit history for these files from TNoodle and into this repository. One simple way of doing that would be to actually start out by making this repo a clone of TNoodle. A fancier approach might be to use `git filter-branch`, which is detailed here: https://gbayer.com/development/moving-files-from-one-git-repository-to-another-preserving-history/

File: README.md (hunk 1 above)
Comment (gregorbg) [reply README.md]: Yes, fixed. They will only be actually removed at the TNoodle main repo once this has been merged upstream (see also my answer to your Q1)

File: build.gradle.kts (hunk 2 above)
Comment (gregorbg) [reply build.gradle.kts]: Fixed!

**Comment by gregorbg** [comment]
Answers to your general questions (same numbering):
1. There will be one as soon as this one is merged upstream. I'd like to avoid situations where the Android sample app is "gone" because the main TNoodle deletion PR gets merged earlier than this one.
2. Yes, will happen as part of the yet-to-come PR mentioned under (1)
3. I have done a very hacky attempt to include a Travis build file. Currently tough because we don't have an actual Maven repository that hosts our main TNoodle artifacts -- so we have to fake them through publication to the local Nexus.
4. I disagree. Most notably, TNoodle **is** a Java/JVM project, so the basic integration examples can and should be based on Java build technology. I have renamed the `tnoodlejs` folder to reflect that opinion. Also, Android is practically married to Gradle. It is unbelievably difficult (if not impossible!) to build Android without Gradle. Lastly, I have no idea how to configure GWT without the convenience of having a Gradle plugin do it for me. We could certainly invest time into improving the situation here, but right now I just want to have these two projects out of the main repo for a bunch of reasons.

**Comment by gregorbg** [comment]
And here's your history! I had to pull in the folder histories for `tnoodle-android` and `tnoodlejs` separately, hence you may see some "duplicated" commits.

In reality, there are two separate merge chains dangling from the tree, where a commit may hve been split up when doing `git filter-branch` on the individual directories. The Internet agrees :tm: that there is no smarter :tm: way to handle this history transfer.

Duplicating the repository does not seem like a viable alternative, because then we would have all of the "unwanted" parts of the code still with us somewhere in the history. So we have to live with merge instead of rebase this one time methinks :)

**Comment by gregorbg** [comment]
Ping @jfly 
I'd like to merge this up within the next 48 hours

**Comment by jfly** [comment]
Sorry for the delay here: LGTM! Thanks for pulling in the history =)

> Lastly, I have no idea how to configure GWT without the convenience of having a Gradle plugin do it for me.

Yeah, it seems totally reasonable to me to use Gradle to configure/invoke GWT. But, I imagine a javascript developer isn't actually interesting in running gradle or gwt, they just want an example of how to `npm install` and use tnoodle. My thinking is that someday... eventually... we'd publish some version of TNoodle on npm for them to be able to use just by running `npm install` locally.

**Pull request merged**